src/AgentCommerce/Subscriber/WebhookSubscriber.php

```diff
@@ -0,0 +1,150 @@
+use Lcobucci\JWT\Encoding\JoseEncoder;
+use Lcobucci\JWT\Signer\Hmac\Sha256;
+use Lcobucci\JWT\Signer\Key\InMemory;
+use Lcobucci\JWT\Token\Builder;
+use Shopware\Core\Framework\Api\Context\AdminApiSource;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityWriteResult;
+use Shopware\Core\Framework\DataAbstractionLayer\Event\EntityWrittenEvent;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Notification\NotificationService;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\System\Country\CountryEntity;
+use Shopware\Core\System\SalesChannel\SalesChannelCollection;
+use Shopware\Core\System\SalesChannel\SalesChannelEntity;
+use Swag\PayPal\Setting\Service\CredentialsUtil;
+use Swag\PayPal\SwagPayPal;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class WebhookSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @param EntityRepository<SalesChannelCollection> $salesChannelRepository
+     */
+    public function __construct(
+        private readonly EntityRepository $salesChannelRepository,
+        private readonly NotificationService $notificationService,
+        private readonly CredentialsUtil $credentialsUtil,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'sales_channel.written' => 'webhook',
+        ];
+    }
+
+    public function webhook(EntityWrittenEvent $event): void
+    {
+        $mapped = [];
+        foreach ($event->getWriteResults() as $writeResult) {
+            /** @var string $id */
+            $id = $writeResult->getPrimaryKey();
+            $active = $writeResult->getProperty('active');
+            if ($active === null || $writeResult->getOperation() === EntityWriteResult::OPERATION_DELETE) {
+                continue;
+            }
+
+            $mapped[$id] = $active;
+        }
+
+        $criteria = new Criteria(array_keys($mapped));
+        $criteria->addFilter(new EqualsFilter('typeId', SwagPayPal::SALES_CHANNEL_TYPE_AGENT_COMMERCE));
+        $criteria->addAssociations([
+            'country',
+            'countries',
+            'currency',
+            'domains',
+            'productExports.storefrontSalesChannel.domains',
+            'productExports.storefrontSalesChannel.hreflangDefaultDomain',
+        ]);
+
+        $result = $this->salesChannelRepository->search($criteria, $event->getContext());
+        foreach ($result as $salesChannel) {
+            if ($mapped[$salesChannel->getId()]) {
+                $this->onboard($salesChannel, $event->getContext());
+            } else {
+                $this->unregister($salesChannel, $event->getContext());
```

Comment:
Because you skip `OPERATION_DELETE`, any deleted sales channel will not be unregistered, right?